src/components/react-hook-form/utils/cancel-button.js

```diff
@@ -1,31 +1,33 @@
-import React from 'react';
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import React, { useMemo } from 'react';
 import { Button } from '@mui/material';
 import { FormattedMessage } from 'react-intl';
 import PropTypes from 'prop-types';
 
-const CancelButton = ({
-    onClick,
-    variant,
-    disabled,
-    withCustomColor = true,
-}) => {
+const CancelButton = ({ withCustomColor = true, ...buttonProps }) => {
+    const colorKey = useMemo(() => {
+        if (withCustomColor) {
+            return 'customButton';
+        } else {
+            return buttonProps?.color ?? 'primary';
+        }
+    }, [withCustomColor, buttonProps?.color]);
     return (
-        <Button
-            onClick={onClick}
-            variant={variant}
-            disabled={disabled}
-            color={withCustomColor ? 'customButton' : 'primary'}
-        >
+        <Button {...buttonProps} color={colorKey}>
             <FormattedMessage id="cancel" />
```

Comment:
add a translation file for this key for this component `cancel-button-fr.js` `cancel-button-en.js`

Reply:
done